StardewArchipelago/Locations/CodeInjections/Modded/MagicModInjections.cs

```diff
@@ -0,0 +1,228 @@
+        private const string LEARN_FIREBALL_AP_LOCATION = "Analyze: Fireball";
+        private const string LEARN_FROSTBITE_AP_LOCATION = "Analyze: Frostbite";
+        private const string LEARN_TELEPORT_AP_LOCATION = "Analyze All Elemental School Locations";
+        private const string LEARN_LANTERN_AP_LOCATION = "Analyze: Lantern";
+        private const string LEARN_TENDRILS_AP_LOCATION = "Analyze: Tendrils";
+        private const string LEARN_PHOTOSYNTHESIS_AP_LOCATION = "Analyze All Nature School Locations";
+        private const string LEARN_DESCEND_AP_LOCATION = "Analyze: Descend";
+        private const string LEARN_METEOR_AP_LOCATION = "Analyze: Meteor";
+        private const string LEARN_LUCKSTEAL_AP_LOCATION = "Analyze: Lucksteal";
+        private const string LEARN_BLOODMANA_AP_LOCATION = "Analyze: Bloodmana";
+        private const string LEARN_REWIND_AP_LOCATION = "Analyze Every Magic School Location";
+
+        private static IMonitor _monitor;
+        private static IModHelper _helper;
+        private static ArchipelagoClient _archipelago;
+        private static LocationChecker _locationChecker;
+
+
+        public static void Initialize(IMonitor monitor, IModHelper modHelper, ArchipelagoClient archipelago, LocationChecker locationChecker)
+        {
+            _monitor = monitor;
+            _helper = modHelper;
+            _archipelago = archipelago;
+            _locationChecker = locationChecker;
+        }
+
+        public static bool OnCast_AnalyzeGivesLocations_Prefix(Farmer player, int level, int targetX, int targetY)
+        {
+            try
+            {
+                if (player != Game1.player)
+                    return false;
+                List<string> spellsLearnedList = new List<string>();
+                if (player.CurrentItem != null)
+                {
+                    if (player.CurrentTool != null)
+                    {
+                        if (player.CurrentTool is StardewValley.Tools.Axe || player.CurrentTool is StardewValley.Tools.Pickaxe)
+                            spellsLearnedList.Add(LEARN_CLEARDEBRIS_AP_LOCATION);
+                        else if (player.CurrentTool is StardewValley.Tools.Hoe)
+                            spellsLearnedList.Add(LEARN_TILL_AP_LOCATION);
+                        else if (player.CurrentTool is StardewValley.Tools.WateringCan)
+                            spellsLearnedList.Add(LEARN_WATER_AP_LOCATION);
+                    }
+                    else if (player.CurrentItem is StardewValley.Objects.Boots)
+                    {
+                        spellsLearnedList.Add(LEARN_EVAC_AP_LOCATION);
+                    }
+                    else if (player.ActiveObject != null)
+                    {
+                        if (!player.ActiveObject.bigCraftable.Value)
+                        {
+                            int index = player.ActiveObject.ParentSheetIndex;
+                            if (index == 395) // Coffee
```

Comment:
If you're aware that these numbers are not obvious and need a comment to explain what they are, why not just make named constants?